backend/app/store/search/store_engine_decorator.go

```diff
@@ -0,0 +1,58 @@
+package search
+
+import (
+	log "github.com/go-pkgz/lgr"
+
+	"github.com/umputun/remark42/backend/app/store"
+	"github.com/umputun/remark42/backend/app/store/engine"
+	service "github.com/umputun/remark42/backend/app/store/search/service"
+)
+
+// StoreEngineDecorator proxies requests to store/engine.Interface and index incoming data
+type StoreEngineDecorator struct {
```

Comment:
This file content is very important part of the code and they should be tested as well.

Reply:
I've added mock for searcher and test with it